### PR TITLE
Increased reload speed

### PR DESCRIPTION
**Pull request by Safort**

### CODE REVIEW

File: src/webpack-plugins/generate-json.ts [hunk 1]
```diff
@@ -3,24 +3,35 @@ import buildBundles from '../scripts/build-bundles';
...
-                Promise.all<any>(fetchRemoteLibs())
-                    .then(() => generateComponentsJSON(false))
-                    .then(callback);
-            } else {
-                callback();
+    partReload = (_, cb) => {
+        cb();
+    }
+
+    beforeCompileCb = process.argv.indexOf('--full-reload') > 0 ? this.fullReload : this.partReload;
```
Comment (Luanre) [inline src/webpack-plugins/generate-json.ts:29]: the argument must be mentioned in README.md

Comment (Safort) [reply src/webpack-plugins/generate-json.ts:29]: Got it.

File: package.json [hunk 2]
```diff
@@ -23,6 +23,7 @@
   "author": "Sberbank",
   "license": "Apache-2.0",
   "dependencies": {
+    "awesome-typescript-loader": "^3.2.3",
```
Comment (Luanre) [inline package.json:26]: you've added `awesome-typescript-loader` but kept `ts-loader`. For what?

File: src/webpack-plugins/generate-json.ts [hunk 3]
```diff
@@ -3,24 +3,35 @@ import buildBundles from '../scripts/build-bundles';
...
 
+export default class GenerateJSON {
     private shouldGenerateJSON: boolean = false;
 
-    apply(compiler) {
-        compiler.plugin('invalid', () => {
-            this.shouldGenerateJSON = true;
-        });
+    fullReload = (_, cb) => {
+        if (this.shouldGenerateJSON) {
```
Comment (Luanre) [inline src/webpack-plugins/generate-json.ts]: I think it's better to use `this.shouldGenerateJSON === true` for a better readability

File: package.json (hunk 2 above)
Comment (Safort) [reply package.json:26]: Forgot to remove this line.

**Pull request merged**